uk.co.compucorp.civicrm.hrcore/hrcore.php

```diff
@@ -26,18 +27,9 @@ function hrcore_civicrm_config(&$config) {
  * @param array $tasks
  */
 function hrcore_civicrm_searchTasks($objectName, &$tasks) {
-  if ($objectName !== 'contact' || !CRM_Core_Permission::check('create users')) {
-    return;
+  if (ContactFormSearchTaskAdder::shouldAdd($objectName)) {
+    ContactFormSearchTaskAdder::add($tasks);
   }
```

Comment:
What about this?

```php
function hrcore_civicrm_searchTasks($objectName, &$tasks) {
  $adders = [
    new AdderOner(),
    new AdderTwo(),
    //....
  ];

  foreach($adders as $adder) {
     if($adder->shouldAdd($objectName)) {
        $adder->addTo($tasks); // Note that I changed the method name to addTo, otherwise, it might look like we're adding the tasks array to the adder
     }
  }
}
```

I don't like the idea of having to instantiate each adder, but I can't think of any other way to do that, given how the hooks work